mala/datahandling/data_handler.py

```diff
@@ -695,14 +666,16 @@ def __load_data(self, function, data_type):
                     units = snapshot.output_units
 
                 if snapshot.snapshot_type == "numpy":
-                    calculator.read_from_numpy_file(file, units=units,
-                                                    array=getattr(self, array)[snapshot_counter])
+                    tmp_array = calculator.read_from_numpy_file(file, units=units)
+                    tmp_array = tmp_array.reshape([gs_new, feature_dimension])
+                    getattr(self, array)[gs_old : gs_old + gs_new] = tmp_array
```

Comment:
The idea of giving the array directly to the function is to save on intermediate memory. By assigning a temporary array, we always have to have extra memory for at least one temporary array. 

What I would suggest is modifying the `read_from_numpy_file` function by adding a reshape option there. Then the same code would work again. Could you try that or else let me know so I will give it a try?

Also could you add the reshape to the OpenPMD section? There we have to bite the bullet with the temporary arrays anyhow because of the way we use OpenPMD.